paasta_tools/check_oom_events.py

```diff
@@ -104,49 +115,38 @@ def latest_oom_events(cluster, superregion, interval=60):
     for e in read_oom_events_from_scribe(cluster, superregion):
         if e["timestamp"] > start_timestamp:
             key = (e["service"], e["instance"])
-            res.setdefault(key, []).append(
-                OOMEvent(
-                    hostname=e.get("hostname", ""),
-                    container_id=e.get("container_id", ""),
-                    process_name=e.get("process_name", ""),
-                )
-            )
+            res.setdefault(key, set()).add(e.get("container_id", ""))
     return res
 
 
-def compose_sensu_status(instance, oom_events, is_check_enabled):
+def compose_sensu_status(
+    instance, oom_events, is_check_enabled, alert_threshold, check_interval
+):
     """
     :param instance: InstanceConfig
     :param oom_events: a list of OOMEvents
     :param is_check_enabled: boolean to indicate whether the check enabled for the instance
     """
+    interval_string = f"{check_interval} minute(s)"
+    instance_name = f"{instance.service}.{instance.instance}"
     if not is_check_enabled:
-        return (
-            Status.OK,
-            "This check is disabled for {}.{}.".format(
-                instance.service, instance.instance
-            ),
-        )
+        return (Status.OK, f"This check is disabled for {instance_name}.")
     if len(oom_events) == 0:
         return (
             Status.OK,
-            "No oom events for %s.%s in the last minute."
-            % (instance.service, instance.instance),
+            f"No oom events for {instance_name} in the last {interval_string}.",
         )
-    else:
+    elif len(oom_events) >= alert_threshold:
         return (
             Status.CRITICAL,
-            "The Out Of Memory killer killed %d processes (%s) "
-            "in the last minute in %s.%s containers."
-            % (
-                len(oom_events),
-                ",".join(
-                    sorted({e.process_name for e in oom_events if e.process_name})
-                ),
-                instance.service,
-                instance.instance,
-            ),
+            f"The Out Of Memory killer killed processes for {instance_name} "
+            f"in the last {interval_string}.",
```

Comment:
It would be great to add some more information here.
Ideally we'd know if the service was using autotune and we would add something like:
If using autotune:
"This service is using autotuned_defaults, its memory requirements will be adjusted automatically at <time of next run of autotune>
If not using autotune:
"Did you know you could let PaaSTA determine how much memory this service need ? Just delete the mem line in your yelpsoa-configs file"

Let's say we log version of docker image as well:
if oom kill alert triggers on version N but wasn't triggered on version N-1
It appears OOM kill started shortly after the release of this new version, maybe the new code has a larger memory footprint.

To remediate this issue, we advise you to manually set the value of the memory you require to a generous value.

Reply:
I would rather update the linked runbook (y/check-oom-events, or a new link for PAASTA-16475) with this information, and keep the message here as short as possible - it's faster to update a runbook than the code.

I don't think the check itself can know which version is newer, and modify the message based on that, but I agree it would be useful to make it easy for service owners to correlate OOMs with version changes and add that tip to the runbook. I'll make a follow-up ticket for that.

Also, the possibility that a service deployment causes OOMs seems like a good reason to still send the alerts to service owners, even if the service is on autotuned_defaults. That way, during the push, the service owner can revert or increase memory as needed, instead of waiting for us to respond.

In general, I don't think we should disable this alerting until we really do respond to OOMs in real time. Until that point, service owners might want to take action based on the OOMs, and I think they should know about them.